integrations/llama_index/src/ucai_llamaindex/toolkit.py

```diff
@@ -169,3 +170,48 @@ def tools(self) -> List[FunctionTool]:
             List[FunctionTool]: A list of tools available in the toolkit.
         """
         return list(self.tools_dict.values())
+
+
+def extract_properties(data: dict) -> dict:
+    """
+    Extracts the 'properties' dictionary from the input dictionary,
```

Comment:
What if the function has a key named `properties`?

Reply:
I think we need an extra layer of validation here, using client to fetch the function info and see if the function input params contains `properties`, and check the data type to see if it matches.

Reply:
Instead of performing a collision check by making another call, I switched to blocking the registration of a uc python function that contains the "properties" field on a registered function. 
I was a bit fed up with the tests, so I've added mocks so this can be locally validated and added unit test equivalents for the e2e tests that require creating functions in UC.